common4j/src/main/com/microsoft/identity/common/java/crypto/BasicEncryptor.java

```diff
@@ -0,0 +1,148 @@
+                          @NonNull final String encryptAlgorithm,
+                          final byte[] iv,
+                          byte[] dataToBeEncrypted) throws ClientException {
+        return performCryptoOperationAndUploadTelemetry(
+                CryptoObjectName.Cipher,
+                encryptAlgorithm,
+                mCryptoFactory,
+                new ICryptoOperation<byte[]>() {
+                    @Override
+                    public byte[] perform() throws ClientException {
+                        return encryptWithCipher(key, encryptAlgorithm, iv, dataToBeEncrypted);
+                    }
+                }
+        );
+    }
+
+    @Override
+    public byte[] encrypt(@NonNull final Key key,
+                          @NonNull final String encryptAlgorithm,
+                          final byte[] iv,
+                          final byte[] dataToBeEncrypted,
+                          final int tagLength,
+                          final byte[] aad) throws ClientException {
+        return performCryptoOperationAndUploadTelemetry(
+                CryptoObjectName.Cipher,
+                encryptAlgorithm,
+                mCryptoFactory,
+                new ICryptoOperation<byte[]>() {
+                    @Override
+                    public byte[] perform() throws ClientException {
+                        return encryptWithCipher(
+                                key, encryptAlgorithm, iv, dataToBeEncrypted, tagLength, aad
+                        );
+                    }
+                }
+        );
+    }
+
+    private byte[] encryptWithCipher(@NonNull final Key key,
+                                     @NonNull final String encryptAlgorithm,
+                                     final byte[] iv,
+                                     byte[] dataToBeEncrypted)
+            throws ClientException {
+        final Cipher cipher = mCryptoFactory.getCipher(encryptAlgorithm);
+        try {
+            if (iv != null && iv.length > 0) {
+                final IvParameterSpec ivSpec = new IvParameterSpec(iv);
+                cipher.init(Cipher.ENCRYPT_MODE, key, ivSpec);
+            } else {
+                cipher.init(Cipher.ENCRYPT_MODE, key);
+            }
+            return cipher.doFinal(dataToBeEncrypted);
+        } catch (final BadPaddingException e) {
+            throw new ClientException(ClientException.BAD_PADDING, e.getMessage(), e);
+        } catch (final IllegalBlockSizeException e) {
+            throw new ClientException(ClientException.INVALID_BLOCK_SIZE, e.getMessage(), e);
+        } catch (final InvalidKeyException e) {
+            throw new ClientException(ClientException.INVALID_KEY, e.getMessage(), e);
+        } catch (final InvalidAlgorithmParameterException e) {
+            throw new ClientException(ClientException.INVALID_ALG_PARAMETER, e.getMessage(), e);
+        }
+    }
+
+    @SuppressWarnings("NewApi")
+    private byte[] encryptWithCipher(@NonNull final Key key,
```

Comment:
encryptWithGcm?

Reply:
Addressed here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1953/commits/8ef8a06a8f8c110c7e06918e087f97256b0ef69c